imports/ui/components/timeline.coffee

```diff
@@ -1,46 +1,117 @@
       if err
         toastr.error(err.message)
         return
+      if myLineChart != null
+        myLineChart.destroy()
       for row in response.data.results
         data = {year: row.year, count: row.count}
         @tld.insert(data)
-      baseYear = @tld.find({}, {sort: {year: -1}}).fetch()[0].year
-      ctryear = baseYear
-      data = {}
-      tdata = @tld.find().fetch()
-      while ctryear > baseYear - 5
-        data[ctryear] = @tld.find({year: ctryear}).fetch()
-        if data[ctryear].length ==  0
-          data[ctryear][0] = {year: ctryear, "count": 0}
-        ctryear--
+      endDate = moment(new Date())
+      baseDate = null
+      ymMax = moment(new Date()).year()
+      monthNames = ["", "January", "February", "March", "April", "May", "June", "July", "August", "September", "October", "November", "December"]
+      xlabels = []
+      counts = []
+      maxCount = 0
+      switch @timelineRange.get()
+        when "6months"
+          endMonth = endDate.month() + 1
+          baseMonth = endDate.subtract(5, 'months')
+          ctr = 0
+          while ctr < 6
+            xlabels.push monthNames[baseMonth.month() + 1]
+            tdata = @tld.find({year: baseMonth.month() + 1}).fetch()
+            if tdata.length == 0
+              counts.push 0
+            else
+              if tdata[0].count > maxCount
+                maxCount = tdata[0].count
+              counts.push tdata[0].count
+            baseMonth.add(1, 'months')
+            ctr++
+        when "1year"
+          endMonth = endDate.month() + 1
+          baseMonth = endDate.subtract(11, 'months')
+          ctr = 0
+          while ctr < 12
+            xlabels.push monthNames[baseMonth.month() + 1]
+            tdata = @tld.find({year: baseMonth.month() + 1}).fetch()
+            if tdata.length == 0
+              counts.push 0
+            else
+              if tdata[0].count > maxCount
+                maxCount = tdata[0].count
+              counts.push tdata[0].count
+            baseMonth.add(1, 'months')
+            ctr++
+        when "5years"
+          endYear = endDate.year()
+          baseYear = endDate.subtract(5, 'years').year()
+          while baseYear <= endYear
+            tdata = @tld.find({year: baseYear}).fetch()
+            if tdata.length == 0
+              counts.push 0
+            else
+              if tdata[0].count > maxCount
+                maxCount = tdata[0].count
+              counts.push tdata[0].count
+            xlabels.push baseYear
+            baseYear++
+        when "all"
+          baseYear = @tld.find({}, {sort: {year: 1}}).fetch()[0].year
+          while baseYear <= endDate.year()
+            tdata = @tld.find({year: baseYear}).fetch()
+            if tdata.length == 0
+              counts.push 0
+            else
+              if tdata[0].count > maxCount
+                maxCount = tdata[0].count
```

Comment:
I would suggest computing this after the counts have all been inserted with `_.max(counts)` so it doesn't need to be repeated in each condition.